src/ngResource/resource.js

```diff
@@ -664,7 +672,9 @@ angular.module('ngResource', ['ng']).
               extend({}, extractParams(data, action.params || {}), params),
               action.url);
 
-            var promise = $http(httpConfig).then(function(response) {
+            var promise = $q.when(requestInterceptor(httpConfig), null, requestErrorInterceptor);
```

Comment:
This will not catch errors thrown in the `requestInterceptor`. The `requestErrorInterceptor` will only be called if `requestInterceptor` returns a rejecting promise.

I think the following is better:

```js
promise.when(httpConfig).then(requestInterceptor).catch(requestErrorInterceptor)
```